src/Storage/FileSystemStorage.php

```diff
@@ -14,11 +14,20 @@
  */
 class FileSystemStorage extends AbstractStorage
 {
-    protected function doUpload(PropertyMapping $mapping, UploadedFile $file, ?string $dir, string $name): ?File
+    protected function doUpload(PropertyMapping $mapping, File $file, ?string $dir, string $name): ?File
     {
         $uploadDir = $mapping->getUploadDestination().\DIRECTORY_SEPARATOR.$dir;
 
-        return $file->move($uploadDir, $name);
+        if ($file instanceof UploadedFile) {
+            return $file->move($uploadDir, $name);
+        } else {
```

Comment:
I guess that here we should check for `ReplacingFile`

Reply:
Resolving, because this is the same case as the resolved comment.